site2/docs/concepts-messaging.md

```diff
@@ -242,6 +242,12 @@ A subscription is a named configuration rule that determines how messages are de
 > * If you want to achieve "message queuing" among consumers, share the same subscription name among multiple consumers(shared, failover, key_shared).
 > * If you want to achieve both effects simultaneously, combine exclusive subscription mode with other subscription modes for consumers.
 
+### Consumerless Subscriptions and Their Corresponding Modes
+If, at any point in time, there are no consumers for a given subscription, a subscription has no specified mode. This
+includes new subscriptions and existing subscriptions that no longer have any consumers. As a consequence, there is no
+option to specify the subscription mode when creating a subscription. The mode is established when the consumer makes its
+connection, and the mode can be "changed" by restarting all consumers with a different configuration.
```

Comment:
Could we refine it as follows?
When a subscription has no consumers, you cannot specify the subscription mode. You can specify the subscription mode when a consumer connects to the subscription, and you can change the subscription mode by restarting all consumers with a different configuration.

Reply:
I'd like to include something that explicitly mentions the following

> if there are no consumers, a subscription has no specified mode.

Based on my reading of your change, it's implicit that a subscription's mode is undefined when there are no consumers, but I'd like it to be explicit. I found the subscription mode's mutability less than intuitive when first trying to understand them. What do you think about the following:

> When a subscription has no consumers, its subscription mode is undefined. A subscription's mode is defined when a consumer connects to the subscription, and the mode can be changed by restarting all consumers with a different configuration.

Reply:
I think it's much better

Reply:
Hi @michaeljmarshall, could you please keep the same style for the other two paragraphs?

Reply:
Hi @michaeljmarshall, could you please keep the same style for the other two paragraphs?

Reply:
I've updated all 3 paragraphs with a new commit. @Jennifer88huang and @Anonymitaet - please let me know what you think.